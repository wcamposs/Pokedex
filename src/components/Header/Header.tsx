// libraries
import React from 'react';
import {
	Platform,
	StyleSheet,
	Text,
	TouchableOpacity,
	View,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { AntDesign } from '@expo/vector-icons';

// js
import colors from '../../utils/colors';

// interfaces
import { HeaderProps } from '../../utils/interfaces';

const Header: React.FC<HeaderProps> = ({
	title,
	showBackButton,
	textColor = colors.red,
}) => {
	const navigation = useNavigation();

	const iconName = Platform.OS === 'ios' ? 'left' : 'arrowleft';

	function handleGoBack() {
		navigation.goBack();
	}

	return (
		<View style={styles.container}>
			<View style={styles.topBar}>
				{!showBackButton ? (
					<View />
				) : (
					<TouchableOpacity onPress={handleGoBack}>
						<AntDesign
							name={iconName}
							size={28}
							color={textColor}
						/>
					</TouchableOpacity>
				)}

				<View style={styles.header}>
					<Text style={styles.title(textColor)}>{title}</Text>
				</View>

				<View />
			</View>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		marginTop: 20,
		padding: 24,
		backgroundColor: colors.transparent,
		zIndex: 100,
		// position: 'absolute',
	},
	topBar: {
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'space-between',
	},
	header: {
		alignItems: 'center',
	},
	title: (textColor: string) => ({
		paddingTop: 12,
		color: textColor,
		letterSpacing: 1,
		fontSize: 32,
		lineHeight: 32,
		textAlign: 'center',
		textTransform: 'uppercase',
		fontWeight: 'bold',
	}),
});

export default Header;
