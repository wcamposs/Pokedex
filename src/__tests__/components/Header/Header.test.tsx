import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../../components/Header/Header';

jest.useFakeTimers();

describe('<Header />', () => {
	it('renders header correctly with title', () => {
		const navigation = jest.fn();

		const tree = renderer
			.create(<Header navigation={navigation} title="Test Title" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders header correctly without title', () => {
		const navigation = jest.fn();

		const tree = renderer
			.create(<Header navigation={navigation} title="" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
