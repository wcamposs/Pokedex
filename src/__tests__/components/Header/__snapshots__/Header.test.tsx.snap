// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders header correctly with title 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "marginTop": 20,
      "padding": 24,
      "zIndex": 100,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#d40000",
            "fontSize": 32,
            "fontWeight": "bold",
            "letterSpacing": 1,
            "lineHeight": 32,
            "paddingTop": 12,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Test Title
      </Text>
    </View>
    <View />
  </View>
</View>
`;

exports[`<Header /> renders header correctly without title 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "marginTop": 20,
      "padding": 24,
      "zIndex": 100,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#d40000",
            "fontSize": 32,
            "fontWeight": "bold",
            "letterSpacing": 1,
            "lineHeight": 32,
            "paddingTop": 12,
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        
      </Text>
    </View>
    <View />
  </View>
</View>
`;
